src/os/bluestore/BlueFS.cc

```diff
@@ -144,7 +144,7 @@ void BlueFS::add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   block_all[id].insert(offset, length);
   block_total[id] += length;
 
-  if (alloc.size()) {
+  if (id < alloc.size() && alloc[id]) {
```

Comment:
reclaim_blocs function below has asserts for both id < alloc.size and alloc[id]. Shouldn't we have similar checking approach for both?